api/maven-api-core/src/main/java/org/apache/maven/api/Constants.java

```diff
@@ -0,0 +1,239 @@
+     */
+    @Config(defaultValue = "auto")
+    public static final String MAVEN_STYLE_COLOR_PROPERTY = "maven.style.color";
+
+    /**
+     * Build timestamp format.
+     *
+     * @since 3.0.0
+     */
+    @Config(source = Config.Source.MODEL, defaultValue = "yyyy-MM-dd'T'HH:mm:ss'Z'")
+    public static final String MAVEN_BUILD_TIMESTAMP_FORMAT = "maven.build.timestamp.format";
+
+    /**
+     * User controlled relocations.
+     * This property is a comma separated list of entries with the syntax <code>GAV&gt;GAV</code>.
+     * The first <code>GAV</code> can contain <code>*</code> for any elem (so <code>*:*:*</code> would mean ALL, something
+     * you don't want). The second <code>GAV</code> is either fully specified, or also can contain <code>*</code>,
+     * then it behaves as "ordinary relocation": the coordinate is preserved from relocated artifact.
+     * Finally, if right hand <code>GAV</code> is absent (line looks like <code>GAV&gt;</code>), the left hand matching
+     * <code>GAV</code> is banned fully (from resolving).
+     * <p>
+     * Note: the <code>&gt;</code> means project level, while <code>&gt;&gt;</code> means global (whole session level,
+     * so even plugins will get relocated artifacts) relocation.
+     * </p>
+     * <p>
+     * For example,
+     * <pre>maven.relocations.entries = org.foo:*:*>, \\<br/>    org.here:*:*>org.there:*:*, \\<br/>    javax.inject:javax.inject:1>>jakarta.inject:jakarta.inject:1.0.5</pre>
+     * means: 3 entries, ban <code>org.foo group</code> (exactly, so <code>org.foo.bar</code> is allowed),
+     * relocate <code>org.here</code> to <code>org.there</code> and finally globally relocate (see <code>&gt;&gt;</code> above)
+     * <code>javax.inject:javax.inject:1</code> to <code>jakarta.inject:jakarta.inject:1.0.5</code>.
+     * </p>
+     *
+     * @since 4.0.0
+     */
+    @Config
+    public static final String MAVEN_RELOCATIONS_ENTRIES = "maven.relocations.entries";
+
+    /**
+     * User property for version filters expression, a semicolon separated list of filters to apply. By default, no version
+     * filter is applied (like in Maven 3).
+     * <p>
+     * Supported filters:
+     * <ul>
+     *     <li>"h" or "h(num)" - highest version or top list of highest ones filter</li>
+     *     <li>"l" or "l(num)" - lowest version or bottom list of lowest ones filter</li>
+     *     <li>"s" - contextual snapshot filter</li>
+     *     <li>"e(G:A:V)" - predicate filter (leaves out G:A:V from range, if hit, V can be range)</li>
+     * </ul>
+     * Example filter expression: <code>"h(5);s;e(org.foo:bar:1)</code> will cause: ranges are filtered for "top 5" (instead
+     * full range), snapshots are banned if root project is not a snapshot, and if range for <code>org.foo:bar</code> is
+     * being processed, version 1 is omitted.
+     * </p>
+     *
+     * @since 4.0.0
+     */
+    @Config
+    public static final String MAVEN_VERSION_FILTERS = "maven.versionFilters";
```

Comment:
This is weird. Here we have `versionFilters`, but above `relocations.entries`. Inconsistent, no?

Reply:
Yes, agreed.